src/scripts/main.js

```diff
@@ -1,7 +1,212 @@
+  return Math.floor(Math.random() * maxVal);
+}
+
+function hasFreeSpace() {
+  return [...refs.fieldCells].some((cell) => cell.textContent === '');
+}
+
+function resetField() {
+  const { messageStart, messageLose, messageWin, score, fieldCells } = refs;
+
+  score.textContent = '0';
+
+  [messageStart, messageLose, messageWin].forEach(({ classList }) =>
+    classList.add('hidden'),
+);
+
+  fieldCells.forEach((cell) => {
+    cell.classList.remove(cell.classList[1]);
+    cell.textContent = '';
+  });
+}
+
+function updateStartButton() {
+  const { startBtn } = refs;
+
+  startBtn.classList.remove('start');
+  startBtn.classList.add('restart');
+  startBtn.textContent = 'Restart';
+}
+
+function moveCells(keyPressed) {
+  updateStartButton();
+
+  !rearrangeCells(keyPressed) && addCell(getRandomCell(), getToken());
+
+  !hasFreeSpace() && !hasPossibleMoves() && finishGame(refs.messageLose);
+}
+
+function addCell(cellPosition, cellValue) {
+  if (cellPosition === -1) {
+    return;
+  }
+
+  const { fieldCells } = refs;
+
+  fieldCells[cellPosition].classList.add(`field-cell--${cellValue}`);
+  fieldCells[cellPosition].textContent = cellValue;
+}
+
+function finishGame(endGameMessage) {
+  endGameMessage.classList.remove('hidden');
+  window.removeEventListener('keydown', inputHandler);
+}
+
+function updateScore(points) {
+  const { score } = refs;
+  const currentPoints = +score.textContent;
+
+  score.textContent = currentPoints + points;
+}
+
+function rearrangeCells(direction) {
+  const { fieldCells, fieldRows } = refs;
+
+  const rows =
+    direction === 'ArrowLeft' || direction === 'ArrowRight'
+      ? [...fieldRows]
+      : createFieldColumns(fieldCells);
+
+  const initialRowValues = rows
+    .map((row) => [...row.children].map((cel) => cel.textContent))
+    .flat();
+
+  rows.forEach((row) => {
+    const initialCells =
+      direction === 'ArrowLeft' || direction === 'ArrowUp'
+        ? [...row.children]
+        : [...row.children].reverse();
+
+    const filledCellsValues = initialCells
+      .filter((cell) => cell.textContent !== '')
+      .map(({ textContent }) => textContent);
+
+    filledCellsValues.forEach((cell, i) => {
+      if (
+        filledCellsValues[i - 1] !== undefined &&
+        filledCellsValues[i - 1] === filledCellsValues[i]
+      ) {
+        updateScore(+filledCellsValues[i] + +filledCellsValues[i - 1]);
+
+        filledCellsValues[i - 1] =
+          +filledCellsValues[i] + +filledCellsValues[i - 1];
+
+        filledCellsValues.splice(i, 1);
+      }
+    });
+
+    const updatedCells =
+      direction === 'ArrowLeft' || direction === 'ArrowUp'
+        ? [...filledCellsValues, '', '', '', ''].slice(0, 4)
+        : [...filledCellsValues, '', '', '', ''].slice(0, 4).reverse();
+
+    [...row.children].forEach((cell, i2) => {
+      const cellValue = updatedCells[i2];
+
+      cell.textContent = cellValue;
+      cell.classList[1] && cell.classList.remove(cell.classList[1]);
```

Comment:
Removing the second class from the classList without checking its value can lead to unintended behavior. Ensure that you're removing the correct class.